core/deployment/src/main/java/io/quarkus/deployment/util/GenericTypeUtil.java

```diff
@@ -0,0 +1,160 @@
+package io.quarkus.deployment.util;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.ParameterizedType;
+import org.jboss.jandex.Type;
+import org.jboss.jandex.TypeVariable;
+
+public final class GenericTypeUtil {
+
+    private static final DotName OBJECT = DotName.createSimple(Object.class.getName());
+
+    public GenericTypeUtil() {
+    }
+
+    /**
+     * Returns the captured generic types of an interface given a class that at some point in the class
+     * hierarchy implements the interface.
+     *
+     * The list contains the types in the same order as they are generic parameters defined on the interface
+     *
+     * A result is only returned iff all the generics where captured. If any of them where not defined by the class
```

Comment:
I'm far from being convinced this is the best behavior. I would have allowed to resolve only the one I'm interested in because there's a good chance I will be interesting in only one of them.

Reply:
TBH I have been pondering this as well. On one hand I absolutely understand the use case of being able to only resolve the parameters I care about, on the other hand it could lead to very surpring behavior where some of the arguments are resolved and others are not.

I'm on the fence with this one so I'll let you decide :)